x/programs/examples/imports/pstate/pstate.go

```diff
@@ -55,15 +56,13 @@ func (i *Import) Register(link runtime.Link, meter runtime.Meter, _ runtime.Supp
 	if err := link.FuncWrap(Name, "get", i.getFn); err != nil {
 		return err
 	}
-	if err := link.FuncWrap(Name, "len", i.getLenFn); err != nil {
-		return err
-	}
 
 	return nil
 }
 
-func (i *Import) putFn(caller *wasmtime.Caller, idPtr int64, keyPtr int32, keyLength int32, valuePtr int32, valueLength int32) int32 {
-	memory := runtime.NewMemory(runtime.NewExportClient(caller))
+func (i *Import) putFn(caller *wasmtime.Caller, idPtr int32, keyPtr int32, valuePtr int32) int32 {
```

Comment:
i think now would be a good time to just move to int64 args which support [len][ptr]. We are creating a fair amount of rework doing this is phases.

I would like to see this signature.
```go
func (i *Import) putFn(caller *wasmtime.Caller, id int64, key int64, value int64) int64 {
   memory := runtime.NewMemory(client)
   idBytes, err := memory.RangeFromInt64(id)
   ...
```

Then extend memory with a method like `memory.RangeFromInt64()`[1] or a helper function.

This ensures we have the full int32 range for the ptr

[1] https://github.com/ava-labs/hypersdk/pull/631/files#diff-46bbbce39f0b86c15aea460cf21fa799bd6fabad826e7ea351c7577e13997ac9R48-R50

Reply:
As I noted I am working on adding direct support so it can look like this.

```
func (i *Import) putFn(caller *program.Caller, keyBytes []byte, valueBytes []byte) int64 {
   id, err :=caller.ID() 
   ...

Reply:
actually, what you mentioned will still be necessary because we are still passing in `ptrs`. I'll update the pr